fhir-server/src/main/java/au/csiro/pathling/fhirpath/literal/QuantityLiteralPath.java

```diff
@@ -6,92 +6,225 @@
     }
+    return decimalValue;
+  }
 
-    return new QuantityLiteralPath(context.getDataset(), context.getIdColumn(), quantity);
+  @Nonnull
+  private static QuantityLiteralPath buildLiteralPath(@Nonnull final BigDecimal decimalValue,
+      @Nonnull final String unit, @Nonnull final Optional<String> display,
+      final @Nonnull FhirPath context, @Nonnull final String fhirPath) {
+    final Quantity quantity = new Quantity();
+    quantity.setValue(decimalValue);
+    quantity.setSystem(Ucum.SYSTEM_URI);
+    quantity.setCode(unit);
+    display.ifPresent(quantity::setUnit);
+
+    return new QuantityLiteralPath(context.getDataset(), context.getIdColumn(), quantity, fhirPath);
   }
 
   @Nonnull
   @Override
   public String getExpression() {
-    return getLiteralValue().getValue().toPlainString() + " '" + getLiteralValue().getUnit() + "'";
+    return expression.orElse(
+        getValue().getValue().toPlainString() + " '" + getValue().getUnit() + "'");
   }
 
+  @Nonnull
+  @Override
+  public Column buildValueColumn() {
+    final Quantity quantity = getValue();
+    final Optional<QuantityComparator> comparator = Optional.ofNullable(quantity.getComparator());
+    final BigDecimal value = quantity.getValue();
+
+    final BigDecimal canonicalizedValue;
+    final String canonicalizedCode;
+    if (quantity.getSystem().equals(Ucum.SYSTEM_URI)) {
+      // If it is a UCUM Quantity, use the UCUM library to canonicalize the value and code.
+      canonicalizedValue = Ucum.getCanonicalValue(value, quantity.getCode());
+      canonicalizedCode = Ucum.getCanonicalCode(value, quantity.getCode());
+    } else if (quantity.getSystem().equals(QuantityLiteralPath.FHIRPATH_CALENDAR_DURATION_URI) &&
+        CALENDAR_DURATION_TO_UCUM.containsKey(quantity.getCode())) {
+      // If it is a (supported) calendar duration, get the corresponding UCUM unit and then use the 
+      // UCUM library to canonicalize the value and code.
+      final String resolvedCode = CALENDAR_DURATION_TO_UCUM.get(quantity.getCode());
+      canonicalizedValue = Ucum.getCanonicalValue(value, resolvedCode);
+      canonicalizedCode = Ucum.getCanonicalCode(value, resolvedCode);
+    } else {
+      // If it is neither a UCUM Quantity nor a calendar duration, it will not have a canonicalized 
+      // form available.
+      canonicalizedValue = null;
+      canonicalizedCode = null;
+    }
+
+    return struct(
+        lit(quantity.getId()).as("id"),
+        lit(value).as("value"),
+        lit(value.scale()).as("value_scale"),
+        lit(comparator.map(QuantityComparator::toCode).orElse(null)).as("comparator"),
+        lit(quantity.getUnit()).as("unit"),
+        lit(quantity.getSystem()).as("system"),
+        lit(quantity.getCode()).as("code"),
+        lit(canonicalizedValue).as(QuantityEncoding.CANONICALIZED_VALUE_COLUMN),
+        lit(canonicalizedCode).as(QuantityEncoding.CANONICALIZED_CODE_COLUMN),
+        lit(null).as("_fid"));
+  }
```

Comment:
Another place, which implicitly depends on the internals of encoders (but so does CodingLiteralPath).
Perhaps it could be moved (at least partially to QuantityEncoder - for the creation of the literal struct).
Or in it's entirety to `QuantityLiteral` if we introduce one (see the comment above)

Reply:
Moved to QuantityEncoder.